src/ui/components/Footer/Footer.react.js

```diff
@@ -25,53 +26,69 @@ export default class Footer extends Component {
     super(props);
   }
 
-  render() {
-    const tracks = this.props.tracks;
-    let status = (tracks !== null) ? utils.getStatus(tracks) : 'An apple a day keeps Dr Dre away';
+  getStatus(props) {
+    const { library } = props;
+    const tracks = library.tracks[library.tracksCursor].sub;
 
+    let status = (tracks !== null) ? utils.getStatus(tracks) : 'An apple a day keeps Dr Dre away';
     const progressBarClasses = classnames('library-refresh-progress', {
       'hidden': !this.props.library.refreshing,
     });
 
-    const library = this.props.library;
-
-    if(library.refreshing) {
+    if (library.refreshing) {
       const progress = Math.round(library.refresh.processed / library.refresh.total * 100);
 
       status = <ProgressBar className={progressBarClasses} now={progress} />;
     }
```

Comment:
can we re-write it so that we don't have to re-assign `status` ?
```js
getStatus(props) {
  const { library } = props;
  const tracks = library.tracks[library.tracksCursor].sub;
  
  const progressBarClasses = classnames('library-refresh-progress', {
    'hidden': !this.props.library.refreshing,
  });

  if (library.refreshing) {
    const progress = Math.round(library.refresh.processed / library.refresh.total * 100);

    return <ProgressBar className={progressBarClasses} now={progress} />;
  } else {
    return tracks !== null ? utils.getStatus(tracks) : 'An apple a day keeps Dr Dre away';
  }
}
```